pytest_operator/plugin.py

```diff
@@ -347,53 +364,184 @@ def download_resource(self, resource, destination: Path):
+        elif not raise_not_found:
+            self._current_alias = None
+        else:
+            raise ModelNotFoundError(f"{alias} not found")
+
+        return self.model
+
+    @property
+    def current_alias(self) -> Optional[str]:
+        return self._current_alias
 
     @property
-    def default_model_name(self):
-        if not hasattr(self, "_default_model_name"):
-            module_name = self.request.module.__name__.rpartition(".")[-1]
-            suffix = "".join(choices(ascii_lowercase + digits, k=4))
-            self._default_model_name = f"{module_name.replace('_', '-')}-{suffix}"
-        return self._default_model_name
-
-    async def run(self, *cmd, cwd=None, check=False, fail_msg=None):
+    def models(self) -> Mapping[str, ModelState]:
+        """Returns the dict of managed models by this fixture."""
+        return {k: dataclasses.replace(v) for k, v in self._models.items()}
+
+    @property
+    def tmp_path(self) -> Path:
+        tmp_path = self._global_tmp_path
+        current_state = self.current_alias and self._models.get(self.current_alias)
+        if current_state and current_state.tmp_path is None:
+            tmp_path = self._tmp_path_factory.mktemp(current_state.model_name)
+            current_state.tmp_path = tmp_path
+        elif current_state and current_state.tmp_path:
+            tmp_path = current_state.tmp_path
+        elif not tmp_path:
+            tmp_path = self._global_tmp_path = self._tmp_path_factory.mktemp(
+                self.default_model_name
+            )
+        log.info(f"Using tmp_path: {tmp_path}")
+        return tmp_path
+
+    @property
+    def model_config(self) -> Optional[dict]:
+        """Represents the config used when adding the model."""
+        current_state = self.current_alias and self._models.get(self.current_alias)
+        return current_state.config if current_state else None
+
+    @property
+    def model(self) -> Optional[Model]:
+        """Represents the current model."""
+        current_state = self.current_alias and self._models.get(self.current_alias)
+        return current_state.model if current_state else None
+
+    @property
+    def model_full_name(self) -> Optional[str]:
+        """Represents the current model's full name."""
+        current_state = self.current_alias and self._models.get(self.current_alias)
+        return current_state.full_name if current_state else None
+
+    @property
+    def model_name(self) -> Optional[str]:
+        """Represents the current model name."""
+        current_state = self.current_alias and self._models.get(self.current_alias)
+        return current_state.model_name if current_state else None
+
+    @property
+    def cloud_name(self) -> Optional[str]:
+        """Represents the current model's cloud name."""
+        current_state = self.current_alias and self._models.get(self.current_alias)
+        return current_state.cloud_name if current_state else None
+
```

Comment:
I could see a case for these raising rather than returning None if current_alias is unset (I don't think I want to continue as normal if `ops_test.cloud_name` returns `None`?) but I think that returning None is in line with how the previous single-model version behaved so this probably makes the most sense

Reply:
yes, most of these were to remain totally backwards compatible and also mypy completeness :)